lib/spack/spack/cmd/bootstrap.py

```diff
@@ -22,84 +22,71 @@
+from spack.fetch_strategy import FetchError
+from spack.package import PackageBase
 
-description = "create a new installation of spack in another prefix"
+from spack.util.executable import which
 
+description = "bootstrap packages needed for spack to run smoothly."
 
 def setup_parser(subparser):
     subparser.add_argument(
-        '-r', '--remote', action='store', dest='remote',
-        help="name of the remote to bootstrap from", default='origin')
+        '-j', '--jobs', action='store', type=int,
+        help="explicitly set number of make jobs. default is #cpus")
     subparser.add_argument(
-        'prefix',
-        help="names of prefix where we should install spack")
-
-
-def get_origin_info(remote):
-    git_dir = join_path(spack.prefix, '.git')
-    git = which('git', required=True)
-    try:
-        branch = git('symbolic-ref', '--short', 'HEAD', output=str)
-    except ProcessError:
-        branch = 'develop'
-        tty.warn('No branch found; using default branch: %s' % branch)
-    if remote == 'origin' and \
-       branch not in ('master', 'develop'):
-        branch = 'develop'
-        tty.warn('Unknown branch found; using default branch: %s' % branch)
-    try:
-        origin_url = git(
-            '--git-dir=%s' % git_dir,
-            'config', '--get', 'remote.%s.url' % remote,
-            output=str)
-    except ProcessError:
-        origin_url = _SPACK_UPSTREAM
-        tty.warn('No git repository found; '
-                 'using default upstream URL: %s' % origin_url)
-    return (origin_url.strip(), branch.strip())
-
-
-def bootstrap(parser, args):
-    origin_url, branch = get_origin_info(args.remote)
-    prefix = args.prefix
-
-    tty.msg("Fetching spack from '%s': %s" % (args.remote, origin_url))
-
-    if os.path.isfile(prefix):
-        tty.die("There is already a file at %s" % prefix)
-
-    mkdirp(prefix)
-
-    if os.path.exists(join_path(prefix, '.git')):
-        tty.die("There already seems to be a git repository in %s" % prefix)
-
-    files_in_the_way = os.listdir(prefix)
-    if files_in_the_way:
-        tty.die("There are already files there! "
-                "Delete these files before boostrapping spack.",
-                *files_in_the_way)
-
-    tty.msg("Installing:",
-            "%s/bin/spack" % prefix,
-            "%s/lib/spack/..." % prefix)
+        '--keep-prefix', action='store_true', dest='keep_prefix',
+        help="don't remove the install prefix if installation fails")
+    subparser.add_argument(
+        '--keep-stage', action='store_true', dest='keep_stage',
+        help="don't remove the build stage if installation succeeds")
+    subparser.add_argument(
+        '-n', '--no-checksum', action='store_true', dest='no_checksum',
+        help="do not check packages against checksum")
+    subparser.add_argument(
+        '-v', '--verbose', action='store_true', dest='verbose',
+        help="display verbose build output while installing")
```

Comment:
> We need to move them into common_arguments since they're used in diy as well.

OK, let's just leave it as-is.  Common arguments should probably go with #2664.